src/systemctl/systemctl-compat-shutdown.c

```diff
@@ -129,8 +130,12 @@ int shutdown_parse_argv(int argc, char *argv[]) {
                         log_error("Failed to parse time specification: %s", argv[optind]);
                         return r;
                 }
-        } else
-                arg_when = now(CLOCK_REALTIME) + USEC_PER_MINUTE;
+        } else {
+                if (logind_has_maintenance_window())
+                        arg_when = USEC_INFINITY; /* logind chooses on server side */
```

Comment:
So I think this deserves some log message of its own, i.e. "Maintainance window scheduled, delaying operation until %s".

maybe implement this by querying the configured scheduled time after you set it. i.e. set a boolean here and then query the value this resulted in later

Reply:
The shutdown commands print that information already via `logind_show_shutdown()`.
```
# shutdown -r
Reboot scheduled for Tue 2023-11-14 11:58:50 CET, use 'shutdown -c' to cancel.
# systemctl reboot --when="13:00"
Reboot scheduled for Tue 2023-11-14 13:00:00 CET, use 'systemctl reboot --when=cancel' to cancel.
```